src/App.test.tsx

```diff
@@ -1,9 +1,33 @@
-import { render, screen } from "@testing-library/react"
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { cleanup, render, screen, waitFor } from "@testing-library/react"
 import App from "./App"
 import React from "react"
+import { usePlayers } from "./hooks/api/usePlayers"
+
+jest.mock("./hooks/api/usePlayers")
+
+const usePlayersMock = jest.mocked(usePlayers)
+
+beforeEach(() => {
+  jest.resetAllMocks()
+})
+
+afterEach(cleanup)
+
+const usePlayersReturn = jest.mocked({
+  hasCurrentSession: jest.fn(),
+  startPollPlayers: jest.fn(),
+})
+
+test("renders register page", async () => {
+  const promise = Promise.resolve(false)
+  usePlayersReturn.hasCurrentSession.mockReturnValue(promise)
+  usePlayersReturn.startPollPlayers.mockReturnValue({ cancel: () => null })
+  // @ts-ignore
+  usePlayersMock.mockReturnValue(usePlayersReturn)
 
-test("renders register page", () => {
   render(<App />)
-  const linkElement = screen.getByText(/Welcome to "Screw your neighbour"/i)
-  expect(linkElement).toBeInTheDocument()
+  await waitFor(() => promise)
```

Comment:
blocks until the second time useEffect ist called in Register.tsx. Then all changes are done and we can end the test.